Add sitemap, robots and manifest

wackerow opened the pull request:
## Description
- Adds `next/sitemap` package and `next-sitemap.config.js` config file
- Config also includes generation of `public/robots.txt` file (added to `.gitignore`)
- Adds `public/manifest.json` with necessary image assets

## Related Issue
https://www.notion.so/efdn/Add-robots-txt-and-sitemap-dbc564c8b7f240a1b5fd81187856bb54?pvs=4

wackerow committed:
yarn add next-sitemap

wackerow committed:
add manifest.json and assets

pettinarip commented on next-sitemap.config.js:
```diff
@@ -0,0 +1,6 @@
+
+module.exports = {
+  siteUrl: process.env.SITE_URL || 'https://ethereum.org',
+  generateRobotsTxt: true,
```
I see that we are indexing the 404 pages
https://deploy-preview-155--ethereum-org-fork.netlify.app/sitemap-0.xml

Maybe we could exclude those from the sitemap generation by using the `exclude` option.

wackerow replied:
Yeah good call, looks like we're doing that on the Gatsby side
<img width="534" alt="image" src="https://github.com/ethereum/ethereum-org-fork/assets/54227730/cfdfe33b-b3fd-4015-966d-c857fecdded8">

Will update

pettinarip replied:
@wackerow I still see some other 404 cases, like the ones for each lang
![image](https://github.com/ethereum/ethereum-org-fork/assets/468158/88cc56ce-360a-4a7c-bce3-32755eb1e6a2)

pettinarip replied:
Fixed.

pettinarip reviewed (commented):
Looks good to me @wackerow 

Comparing it with the prod version, I see that we are changing the `changefreq` to `daily` and the `priority` to `0.5`. In prod we were using `monthly` & `0.7`. Not sure if this might have a negative impact or not, just wanted to call it out.

cc @nhsz not sure if you have any preference here

wackerow committed:
Merge branch 'dev' into sitemap-robots-manifest

wackerow commented:
> Comparing it with the prod version, I see that we are changing the changefreq to daily and the priority to 0.5. In prod we were using monthly & 0.7. Not sure if this might have a negative impact or not, just wanted to call it out.

Looks like in prod we're using `weekly`/`0.7` for _English_, and `monthly`/`0.5` for _non-English_. This plugin defaults to `daily`/`0.7`... I'll look into using a transform to add a conditional here to match parity with Gatsby site.

wackerow committed:
add sitemap transform

exclude 404, fix changefreq and priority depending on English vs non-English paths

pettinarip committed:
exclude all 404 pages from sitemap generation

pettinarip committed:
make 404 check stricter

wackerow commented:
Another option here that I left off was the `trailingSlash`... This has not been added to our `tsconfig` although we may want to consider; the canonical repo used this and if we plan to deploy a static export to IPFS I think this will be required. Doesn't have to block this PR, can discuss more.

nhsz reviewed (approved):
lgtm
